### ISSUE TITLE
when closing stale PRs, link to an explanation of our PR queue practices

### ISSUE THREAD

**Issue by SethTisue** [scala/scala-dev#577]
`README.md` or `CONTRIBUTING.md` should have an explanation we can link to, in case someone is curious

as per discussion starting at https://gitter.im/scala/contributors?at=5be19ca96494cd16731f7e3b

**Comment by dwijnand** [comment]
Good idea.  I would suggest a page in https://github.com/scala/scala-dev/wiki (to avoid having a wiki in the scala/scala repo), much like the community build uses the GitHub wiki, but then you'd have to change the policy stated in https://github.com/scala/scala-dev/wiki/Home.

**Comment by soronpo** [comment]
I would to chime in with a proposal. The trouble with closing such stale PRs is that they are still different from PRs closed due to other reasons such as invalidity. I propose one of the following:
1. Flag the stale PRs that are closed in some way (`stale`, `zombie`, or other name).
2. Don't close those PRs, but create a new milestone named WIP and assign all those stale PRs to that milestone. I've seen this approach done in other repos.

**Comment by dwijnand** [comment]
I'm fairly sure the Scala Team would be fine with labelling them as stale for reaping reasons.

**Comment by SethTisue** [comment]
Dotty repo uses a "t:revisit" label.

**Comment by som-snytt** [comment]
I just happened to notice an old label, "needs attention," which I would use all the time, obviously. Maybe shorten it to "needy".

**Comment by SethTisue** [comment]
I'm against any solution that involves keeping abandoned PRs open indefinitely. (I'm prepared to argue the point if necessary.)

I think I like the label idea. I'm not sure about `t:revisit`... I don't want to imply that the core team intends to revisit it (as opposed it being a community, volunteer matter).

`stale` is maybe okay, but I'm not sure it communicates the right thing. `zombie` is even more unclear.

But I'm not sure I have a better idea than `stale`. I want to communicate "this was set aside because activity lapsed, but it could be revived any time, either by the original submitter, or by any interested person", not sure how to communicate that in a word or two 🤷. "suspended"? "paused"? "orphaned"? "abandoned"? "dormant"? "in limbo"? ("on hiatus"? works for rock bands and TV shows, but probably doesn't work here)

(hmm, I kinda like "dormant", I think? idk, though)

**Comment by bjornregnell** [comment]
perhaps "undecided"

**Comment by janekdb** [comment]
shelved? Git inspired.

**Comment by SethTisue** [comment]
My main concern is not to be too negative. We want to encourage, rather than discourage, resumption of work on the PR. Closing a PR already runs the risk of being discouraging, which is why this ticket exists.

That's my main reservation about most of the possible choices here.

I keep trying to come up with something like "candidate-for-resurrection" or "needs-new-owner" or "adopt-me" or "resumption-welcome" or.... something

I'm probably trying too hard to make a label name do so much work...

**Comment by SethTisue** [comment]
"shelter-puppy"

**Comment by bjornregnell** [comment]
That's why i think "undecided" might be god. It's not negative, just say "we don't know enough yet".

**Comment by bjornregnell** [comment]
"revisit" as in dotty is also not negative, and perhaps more encouraging than "undecided"

**Comment by janekdb** [comment]
available?

Janek

**Comment by som-snytt** [comment]
`tabled`? More furniture inspiration.

Note that "tabled" means neither "on the table" nor "off the table".

Oh, wikipedia explains: "In parliamentary procedure, the verb to table has the opposite meaning in different countries."

So that makes it especially useful for a PR label intended to be multivalent.

**Comment by som-snytt** [comment]
> stale for reaping reasons.

Another suitably ambiguous label: `ripe for reaping`.

Does that mean "ready for review" or "the smell of bit rot demands that we close this PR".

Someone will either say looks good or just close it and open a window.

**Comment by joroKr21** [comment]
Orphaned?

**Comment by som-snytt** [comment]
Seth's list includes "orphaned", capturing the vibe in the room. I also like his suggestion ":shrug:".

**Comment by joroKr21** [comment]
deserted, forsaken, forgotten, forlorn, disowned, vacant

**Comment by martijnhoekstra** [comment]
Any of Seths suggestions sounds good to me. When a PR closes as stale, a comment explaining what exactly should still happen to bring it to fruition might be more important than the tag.

**Comment by som-snytt** [comment]
The older idiom for putting a WIP into a dormant state was to call it a SIP.

**Comment by michelou** [comment]
> The older idiom for putting a WIP into a dormant state was to call it a SIP.

Or maybe simply R.I.P.

**Comment by Jasper-M** [comment]
> Or maybe simply R.I.P.

That's what we call a Scala 3 SIP.

**Comment by som-snytt** [comment]
> available?

`waiting for you`

taking inspo from the sex bots of twitter.
